### ISSUE TITLE
Session is killed while initializing the fresh parameters for the chatbot

### ISSUE THREAD

**Issue by shresthpaul133** [chiphuyen/stanford-tensorflow-tutorials#98]
This is the trouble I'm facing when I run the chatbot.py file. Any idea on what should I do. When I downloaded the file and I'm running it this is what the error is coming up again and again. 
This is coming on the Terminal. 

Bucketing conversation number 129999
Bucketing conversation number 139999
Bucketing conversation number 149999
Bucketing conversation number 159999
Number of samples in each bucket:
 [5972, 13166, 9429, 15926, 16756, 14772]
Bucket scale:
 [0.0785572407624209, 0.251746228015943, 0.37577774562292, 0.5852724904960471, 0.80568527117507, 1.0]
Initialize new model
Create placeholders
Create inference
Creating loss... 
It might take a couple of minutes depending on how many buckets you have.
WARNING:tensorflow:From /home/shresth/.local/lib/python3.5/site-packages/tensorflow/python/ops/nn_impl.py:1346: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See tf.nn.softmax_cross_entropy_with_logits_v2.

Time: 51.51488280296326
Create optimizer... 
It might take a couple of minutes depending on how many buckets you have.
Creating opt for bucket 0 took 10.933812856674194 seconds
Creating opt for bucket 1 took 16.826176643371582 seconds
Creating opt for bucket 2 took 19.97692561149597 seconds
Creating opt for bucket 3 took 25.283050537109375 seconds
Creating opt for bucket 4 took 31.175849437713623 seconds
Creating opt for bucket 5 took 35.5855815410614 seconds
Running session
Initializing fresh parameters for the Chatbot
Killed

**Comment by MartinAbilev** [comment]
once i got the same warning and crash ... i can`t remmember what i did exactly ... but you need to get rid of this warning !!! what is version of python you using ??

**Comment by andrediasesp** [comment]
Getting the same warning, does anyone know how to fix it?

**Comment by shresthpaul133** [comment]
@MartinAbilev Can you give me your code. What changes you made. I really need it. Please

**Comment by MartinAbilev** [comment]
https://github.com/MartinAbilev/tenzorflow-chat-testdrive it is one of multiple versions i got.  it is working for me at this moment   it is on python3

**Comment by chiphuyen** [comment]
The warning shouldn't affect its behavior. What platform are you running the code on?

**Comment by andrediasesp** [comment]
@chiphuyen i'm running it on Linux, i think that's a memory problem. The OS is killing it .

**Comment by shresthpaul133** [comment]
@chiphuyen Mam I'm using Ubuntu, and mu laptop's ram is 4GB. And it's pretty fast on Linux. It's not that fast on Windows. Mam I even inserted numbers at every point of code. To see exactly where the error is coming. The error is in the 
def run_step function, and in particular 

if not forward_only:
        output_feed = [model.train_ops[bucket_id],  # update op that does SGD.
                       model.gradient_norms[bucket_id],  # gradient norm.
                       model.losses[bucket_id]]  # loss for this batch.
    else:
        output_feed = [model.losses[bucket_id]]  # loss for this batch.
        for step in range(decoder_size):  # output logits.
            output_feed.append(model.outputs[bucket_id][step])

    outputs = sess.run(output_feed, input_feed)
    if not forward_only:
        return outputs[1], outputs[2], None  # Gradient norm, loss, no outputs.
    else:
        return None, outputs[0], outputs[1:]  # No gradient norm, loss, outputs.

This part of the code is where I'm getting stuck.
Please help and can you tell me when you are also free so that I can also come online at that time and can solve the problem asap.
